src/storage-client/src/types/sources.rs

```diff
@@ -1250,6 +1259,7 @@ impl UnplannedSourceEnvelope {
         key_desc: Option<RelationDesc>,
         value_desc: RelationDesc,
         metadata_desc: RelationDesc,
+        order_by: Option<String>,
```

Comment:
Rather than passing in the `order_by` option here, I think you want to adjust `UnplannedSourceEnvelope::Upsert` to `UnplannedSourceEnvelope::Upsert { order_by_column: Option<usize> }`.